### PR TITLE
Internationalization for Questionnaire AnswerOptions

### LINKED ISSUE #929: Internationalisation for the Questionnaire answerOption item using Translation Extensions
**Is your feature request related to a problem? Please describe.**
We currently do not have internationalisation for the [Questionnaire item answer options](https://www.hl7.org/fhir/questionnaire-definitions.html#Questionnaire.item.answerOption) 

**Describe the solution you'd like**
Support for multiple translations can be added using _Extensions_ by implementing the [HL7 FHIR spec here](http://hl7.org/fhir/languages.html##ext) _(Translation Extensions)_

**Describe alternatives you've considered**
The alternatives for translations include `CodeSystem` and `ValueSet` designations as defined in [the spec here](http://hl7.org/fhir/languages.html##term). Those will be implemented/tracked as a separate ticket here #552 

**Additional context**
We already have _Translation Extensions_ implemented on the FHIR SDK for the `QuestionnaireItemComponents`. See #465

**Would you like to work on the issue?**
Yeah

### PR DESCRIPTION
**Pull request by ndegwamartin**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #929

**Description**
  Add Multi language translations support for Questionnaire Answer Options via FHIR Extensions
. 

**Alternative(s) considered**
Alternatives for Multi language support are CodeSystems and ValueSets via Designations. See #929 for context.

**Type**
Choose one: (Feature)

**Screenshots (if applicable)** 

```
                    "answerOption": [
                        {
                            "valueCoding": {
                                "system": "http://hl7.org/fhir/administrative-gender",
                                "code": "female",
                                "display": "Female",
                                "_display": {
                                    "extension": [
                                        {
                                            "url": "http://hl7.org/fhir/StructureDefinition/translation",
                                            "extension": [
                                                {
                                                    "url": "lang",
                                                    "valueCode": "sw-KE"
                                                },
                                                {
                                                    "url": "content",
                                                    "valueString": "Mwanamke"
                                                }
                                            ]
                                        }
                                    ]
                                }
                            }
                        },
```

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreAnswerOptionsTest.kt [hunk 1]
```diff
@@ -53,4 +57,48 @@ class MoreAnswerOptionsTest {
...
+          }
+      }
+
+    Locale.setDefault(Locale.forLanguageTag("vi-VN"))
+
+    assertThat(answerOption.displayString).isEqualTo("Thí nghiệm")
+  }
+
+  @Test
+  fun getDisplayString_choiceItemType_answerOptionShouldReturnDisplayValueForMalformedLanguageExtension() {
```
Comment (santosh-pingle) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/MoreAnswerOptionsTest.kt:84]: is it ok to rename it as
```
getDisplayString_invalidExtension_shouldReturnDisplayValue

```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreAnswerOptionsTest.kt [hunk 2]
```diff
@@ -53,4 +57,48 @@ class MoreAnswerOptionsTest {
...
+            display = "Test Code"
+            addExtension(
+              Extension(ToolingExtensions.EXT_TRANSLATION).apply {
+                addExtension(Extension("lang", StringType("vi-VN")))
+                addExtension(Extension("content", StringType("Thí nghiệm")))
+              }
+            )
+          }
+      }
+
```
Comment (santosh-pingle) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/MoreAnswerOptionsTest.kt:77]: can we remove extra blank line?

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreAnswerOptionsTest.kt [hunk 3]
```diff
@@ -53,4 +57,48 @@ class MoreAnswerOptionsTest {
...
+            display = "Test Code"
+            addExtension()
+            addExtension(
+              Extension(ToolingExtensions.EXT_TRANSLATION).apply {
+                addExtension(Extension("lang", StringType("vi-VN")))
+              }
+            )
+          }
+      }
+
```
Comment (santosh-pingle) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/MoreAnswerOptionsTest.kt:99]: can we remove extra blank line?

**Review by santosh-pingle — COMMENTED** [review]
left some comments.

**Comment by ndegwamartin** [comment]
@santosh-pingle made the updates based on your feedback, please have a look

**Review by jingtang10 — CHANGES_REQUESTED** [review]
Please see the comment I made in the linked issue https://github.com/google/android-fhir/issues/929#issuecomment-987197522

According to the example I found here: https://build.fhir.org/ig/HL7/fhir-ips/StructureDefinition-Coding-uv-ips.html it seems the translation extension is on the `display` (which is a string) rather than the `Coding`.

If that's the case, then all we need to do is to reuse the extension function

```
fun StringType.getLocalizedText
```

I would suggest moving that function to the `common` module if that's the path we decide to go down.

**Comment by Tarun-Bhardwaj** [comment]
@ndegwamartin , though you have committed changes 3 days back, the status still shows 'Changes requested'. Is there an ETA by when the remaining changes can be committed?

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 4]
```diff
@@ -91,7 +91,7 @@ object Dependencies {
     const val runtime = "androidx.room:room-runtime:${Versions.Androidx.room}"
   }
 
-  const val androidFhirCommon = "com.google.android.fhir:common:${Versions.androidFhirCommon}"
```
Comment (jingtang10) [inline buildSrc/src/main/kotlin/Dependencies.kt:94]: revert

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 5]
```diff
@@ -168,7 +168,7 @@ object Dependencies {
       const val stdlib = "1.5.31"
     }
 
-    const val androidFhirCommon = "0.1.0-alpha01"
+    const val androidFhirCommon = "0.1.0-alpha01-preview3-SNAPSHOT"
```
Comment (jingtang10) [inline buildSrc/src/main/kotlin/Dependencies.kt:171]:
```suggestion
    const val androidFhirCommon = "0.1.0-alpha02"
```

**Review by jingtang10 — CHANGES_REQUESTED** [review]
https://github.com/google/android-fhir/pull/1037 has been released today.

looks good. just have to update the dependency.

**Comment by ndegwamartin** [comment]
> 

@Tarun-Bhardwaj The PR was awaiting the Alpha 02 Release of the FHIR Common Library which is now ready. I've updated the PR

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**